internal/store/engine_state.go

```diff
@@ -254,8 +255,8 @@ type ManifestState struct {
 	ConfigFilesThatCausedChange []string
 }
 
-func NewState() *EngineState {
-	ret := &EngineState{}
+func NewState(b model.TiltBuild) *EngineState {
```

Comment:
hmmm...it seems weird to invent a new way to pass in initial data.

have you considered using InitAction?

it might also make sense to inject this data into SailClient and have it pass it up directly; i could see arguments before or against putting this in engine state.

Reply:
i _did_ consider initAction, went with this instead but not for any particular reason, just figured i already had a `tiltBuild` floating around during wiring and might as well use it. can change.

if you feel strongly i can def put this data straight onto SailClient -- but it's not really a property OF the SailClient, more of the run as a whole, which is why my instinct was EngineState.